### ISSUE TITLE
NameError: global name 'method' is not defined

### ISSUE THREAD

**Issue by AndreasEisele** [peshay/btcde#10]
Dear all,

I was very happy to discover this module, which I am quite eager to try out.
Thanks a lot for writing and making it available!

But my first attempt failed. I followed the steps in the README.md up to this point:

```
>>> orderbook = btcde.showOrderbook(conn, 'buy')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "btcde.py", line 137, in showOrderbook
    return APIConnect(conn, 'GET', params, orderuri)
  File "btcde.py", line 115, in APIConnect
    header = set_header(url)
  File "btcde.py", line 84, in set_header
    if method == 'POST':
NameError: global name 'method' is not defined
```
Looks manageable, but I thought you might want to know about this.

Best regards from Luxembourg,
Andreas

**Comment by peshay** [comment]
Hi Andreas! You just come up with this at the right time! :)
Because I was working on some testing code and also found issues and made this module also compatible to v2 API by bitcoin.de
But currently I am not able to do tests with bitcoin.de on my own, can you please test my current branch here: https://github.com/peshay/btcde/tree/get_test_coverage
If it works as expected I can release it as a new version.
Thanks
Andreas

**Comment by AndreasEisele** [comment]
Hi again, I'll be glad to help a bit by testing.

So far, I found one small problem: As I did not know what the valid trading
pairs are, I tried an invalid one. The error message was not very helpful,
looks more like a bug in the code:

>>> orderbook = btcde.showOrderbook(conn, 'buy', 'btc/eur')
problem
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "btcde.py", line 138, in showOrderbook
    params.update(args)
UnboundLocalError: local variable 'params' referenced before assignment

With a valid pair, it looks better:

>>> orderbook = btcde.showOrderbook(conn, 'buy', 'btceur')
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1):
api.bitcoin.de
DEBUG:urllib3.connectionpool:https://api.bitcoin.de:443 "GET
/v2/orders?trading_pair=btceur&type=buy HTTP/1.1" 200 None
>>>

That's all I can say so far. I'll try more stuff and keep you up to date.

Viele Grüße, Andreas
